js/client-ladder.js

```diff
@@ -135,22 +136,28 @@
 				var self = this;
 				this.$el.html('<div class="ladder pad"><p><button name="selectFormat"><i class="fa fa-chevron-left"></i> Format List</button></p><p><em>Loading...</em></p></div>');
 				if (app.localLadder) {
-					app.send('/cmd laddertop ' + format);
+					app.send('/cmd laddertop ' + format + (this.curSearchVal ? ' ,' + this.curSearchVal : ''));
 				} else {
 					$.get('/ladder.php', {
 						format: format,
 						server: Config.server.id.split(':')[0],
-						output: 'html'
+						output: 'html',
+						prefix: this.curSearchVal
 					}, function (data) {
 						if (self.curFormat !== format) return;
-						var buf = '<div class="ladder pad"><p><button name="selectFormat"><i class="fa fa-chevron-left"></i> Format List</button></p><p><button class="button" name="refresh"><i class="fa fa-refresh"></i> Refresh</button></p>';
-						buf += '<h3>' + BattleLog.escapeFormat(format) + ' Top 500</h3>';
+						var buf = '<div class="ladder pad"><p><button name="selectFormat"><i class="fa fa-chevron-left"></i> Format List</button></p><p><button class="button" name="refresh"><i class="fa fa-refresh"></i> Refresh</button>';
+						buf += '<form><input type="text" id="ladderSearchBar" name="search" class="textbox searchinput" value="' + (self.curSearchVal || '') + '" placeholder="username prefix"/><button name="search" type="submit"> Search</button></form></p>';
+						buf += '<h3>' + BattleLog.escapeFormat(format) + ' Top ' + (self.curSearchVal ? "- '" + self.curSearchVal + "'" : '500') + '</h3>';
 						buf += data + '</div>';
 						self.$el.html(buf);
 					}, 'html');
 				}
 			}
 		},
+		search: function (e) {
```

Comment:
```suggestion
		submitSearch: function (e) {
```

This will need to be wired up in the events table as `"submit .search": "submitSearch"`